src/main/java/uk/ac/sanger/sccp/stan/service/PermServiceImp.java

```diff
@@ -48,13 +49,15 @@ public OperationResult recordPerm(User user, RecordPermRequest request) {
         requireNonNull(request, "Request is null");
         final Set<String> problems = new LinkedHashSet<>();
         Labware lw = lookUpLabware(problems, request.getBarcode());
+        Labware controlLabware = lookUpControlLabware(problems, request.getPermData());
```

Comment:
Is it required to check null values returned for lw and controlLabware so as to avoid following checks?

Reply:
Whether or not the control labware is null is checked in several places in the service.